### PR TITLE
RAC-879: Clean remove attribute maintenance command does not unblacklisted attributes

### PR DESCRIPTION
**Pull request by SamirBoulil**
**Description (for Contributor and Core Developer)**

Rework the clean attribute command by:
- Removing the possiblity to "refresh" all the catalog products
- Add the possibility to run a clean job for all blacklisted attributes at once with the --all option.
- Checks if the given attribute codes are indeed blacklisted otherwise, explains the user that the given attribute code does not exist in the blacklist.

Screen Shot 2021-09-29 at 17 19 03

**Definition Of Done (for Core Developer only)**

- [x] Tests

### CODE REVIEW

**Review by ghoussard — APPROVED** [review]

File: src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php [hunk 1]
```diff
@@ -50,6 +47,7 @@ class CleanRemovedAttributesFromProductAndProductModelCommand extends Command
     private CountProductsWithRemovedAttributeInterface $countProductsWithRemovedAttribute;
     private CountProductModelsWithRemovedAttributeInterface $countProductModelsWithRemovedAttribute;
     private CountProductsAndProductModelsWithInheritedRemovedAttributeInterface $countProductsAndProductModelsWithInheritedRemovedAttribute;
+    private FindBlacklistedAttributesCodesInterface $findBlacklistedAttributesCodes;
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php:50]: There is this query on `master` https://github.com/akeneo/pim-community-dev/blob/master/src/Akeneo/Pim/Structure/Bundle/Query/InternalApi/Attribute/GetAllBlacklistedAttributeCodes.php#L10
Will we replace it with your query or should we backport it instead?

File: src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php [hunk 2]
```diff
@@ -90,7 +90,13 @@ protected function configure()
     {
         $this
             ->setDescription('Removes all values of deleted attributes on all products and product models')
-            ->addArgument('attributes', InputArgument::OPTIONAL | InputArgument::IS_ARRAY);
+            ->addArgument('attributes', InputArgument::OPTIONAL | InputArgument::IS_ARRAY)
+            ->addOption(
+                'all',
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php:95]: Also different naming on master https://github.com/akeneo/pim-community-dev/blob/master/src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php#L97

File: src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php [hunk 3]
```diff
@@ -101,98 +107,50 @@ protected function execute(InputInterface $input, OutputInterface $output): int
...
-            $productToCleanCount++;
-            if (0 === $productToCleanCount % $productBatchSize) {
-                $this->launchCleanTask($productIds, $env, $rootDir);
-                $cacheClearer->clear();
-                $productIds = [];
-
-                $progressBar->advance($productBatchSize);
-            }
-        }
-        if (count($productIds) > 0) {
```
Comment (ValentinMumble) [inline src/Akeneo/Pim/Enrichment/Bundle/Command/CleanRemovedAttributesFromProductAndProductModelCommand.php:181]: I believe we should keep this behaviour of cleaning all products as a last resort, and also unblacklist attribute afterwards

**Comment by SamirBoulil** [comment]
Reworked in https://github.com/akeneo/pim-community-dev/pull/15399

**Pull request closed without merging**